src/main/java/org/apache/commons/lang3/tuple/ArchUtilsImproved.java

```diff
@@ -0,0 +1,400 @@
+     * </p>
+     *
+     * @return <code>True</code> if the current JVM is a PPC JVM, else <code>false</code>.
+     */
+    public static final boolean isPPCJVM() {
+        return isPPCJVM(SystemUtils.OS_ARCH);
+    }
+
+    /**
+     * Checks if the given {@link String} represents a PPC JVM. The {@link String} must be
+     * like a value returned by the os.arch System Property.
+     *
+     * @param value The value to check.
+     * @return <code>True</code>, if the {@link String} represents a PPC value, else <code>false</code>.
+     */
+    public static final boolean isPPCJVM(String value) {
+        return is32BitJVM(value) && ProcessorType.PPC.equals(map.get(value).getProcessorType());
+    }
+
+    /**
+     * Checks if the current running JVM is a JVM for PPC 64 bit architecture.
+     * It returns <code>true</code>, if the os.arch System Property matches the following {@link String}'s:
+     * <ul>
+     * <li>ppc64</li>
+     * <li>powerpc64</li>
+     * <li>power_pc64</li>
+     * <li>power_rs64</li>
+     * </ul>
+     * <p>
+     * It is possible to extend the {@link String}'s above using method {@link ArchUtils#addPPC64(String)}.
+     * </p>
+     * <p>
+     * Important: The os.arch System Property returns the architecture used by the JVM
+     * not of the operating system.
+     * </p>
+     *
+     * @return <code>True</code> if the current JVM is a PPC 64 bit JVM, else <code>false</code>.
+     */
+    public static final boolean isPPC64JVM() {
+        return isPPC64JVM(SystemUtils.OS_ARCH);
+    }
+
+    /**
+     * Checks if the given {@link String} represents a PPC 64 JVM. The {@link String} must be
+     * like a value returned by the os.arch System Property.
+     *
+     * @param value The value to check.
+     * @return <code>True</code>, if the {@link String} represents a PPC 64 value, else <code>false</code>.
+     */
+    public static final boolean isPPC64JVM(String value) {
+        return is64BitJVM(value) && ProcessorType.PPC.equals(map.get(value).getProcessorType());
+    }
+
+    /**
+     * Checks if the current running JVM is a 32 bit JVM using the os.arch System Property.
+     * <p>
+     * Important: The os.arch System Property returns the architecture used by the JVM
+     * not of the operating system.
+     * </p>
+     *
+     * @return <code>True</code>, if the current JVM is 32 bit, else <code>false</code>.
+     */
+    public static final boolean is32BitJVM() {
+        return is32BitJVM(SystemUtils.OS_ARCH);
+    }
+
+    /**
+     * Checks if the given {@link String} represents a 32 bit JVM. The {@link String} must be
+     * like a value returned by the os.arch System Property.
+     *
+     * @param value The value to check.
+     * @return <code>True</code>, if the {@link String} represents a 32 bit value, else <code>false</code>.
+     */
+    public static final boolean is32BitJVM(String value) {
+        if (isSupported(value)) {
+            Processor processor = map.get(value);
```

Comment:
Unnecessary check

Could just code the method as:

return ProcessorArch.BIT_32.equals(processor.getProcessorArch());